### PR TITLE
Update nested primkey

### PR DESCRIPTION
**Pull request by Fridus**
This is an improvement over the other MR #67 

-  Controller `update` use the controller `read` to return data. (More logical)  
@mbroadst : Controller `read` and `list` are unmodified. They always returned the associated data if this is defined on resource.
- Allow change associated object based on nested identifier
capture d ecran 2015-06-04 a 15 57 07
- Reload the instance before send it if needed
- I think this is not a breaking change because the identifier is present like before this feature.

### CODE REVIEW

File: lib/Controllers/update.js [hunk 1]
```diff
@@ -47,10 +28,45 @@ Update.prototype.write = function(req, res, context) {
       context.attributes[a] = req.params[a];
   });
 
+  // Update associated data
+  var self = this,
+      associationPrimKeys = [];
+  if (this.include && this.include.length) {
+    this.include.forEach(function(i) {
+      var instanceModel = i.model || i,
```
Comment (johngiudici) [inline lib/Controllers/update.js]: I personally feel that this should be transformed when the resource is built, so that if the user provides something like `include: [SomeModel]`, epilogue will automatically change that to `include: [{ model: SomeModel }]`. We're going to be referencing includes more and more, and it would be nice to not have to do this check every time.

Comment (Fridus) [inline lib/Controllers/update.js]: Replace https://github.com/dchester/epilogue/blob/168332280ef0bd5d5660c59552e4f4a1075a285b/lib/Resource.js#L21 by 
```
  this.include = options.include.map(function(include) {
    if (!include.model) {
      include = { model: include };
    }
    return include;
  });
```
 ?

Tested and it work. I can add this on this MR. @mbroadst

Comment (johngiudici) [inline lib/Controllers/update.js]: I would personally check to see if include is an instance of a Sequelize model. Also, this should have a test. [See here](https://github.com/johngiudici/epilogue/commit/c46138c55f4dd8a0b3c197f46c20b811b9851a50). Maybe this should be a separate PR if we all agree it's a good idea?

Comment (mbroadst) [inline lib/Controllers/update.js]: yeah separate PR, I think we all can agree this is a good idea :smile:

Comment (Fridus) [inline lib/Controllers/update.js]: :+1:

Comment (Fridus) [inline lib/Controllers/update.js]: Now that the MR has been accepted, may be I can rebase this branch and do not check if the model is present in include.
> var instanceModel = i.model || i

**Comment by Fridus** [comment]
@mbroadst Ready to merge ? :)
